### PR TITLE
chore: add `.pre-commit-hooks.yaml`

### PR DESCRIPTION
**Pull request by 34j**
This PR adds a [pre-commit](https://pre-commit.com/#new-hooks) hook.

### CODE REVIEW

**Comment by jeffa5** [comment]
Can you say why we need this?

**Comment by 34j** [comment]
Sorry for your confusion, you could refer to my explanation in an similar PR for another typst formatter:

https://github.com/Enter-tainer/typstyle/pull/91#issuecomment-2162852338

**Comment by 34j** [comment]
pre-commit builds it automatically as I specified `cargo`, so there is no need for the user to install it. Although I am not sure what specific commands are used.

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- id: typstfmt
+  name: typstfmt
+  description: Format files with typstfmt
+  entry: typstfmt
+  language: rust
```
Comment (34j) [inline .pre-commit-hooks.yaml:5]: Here

**Comment by 34j** [comment]
image
image

**Pull request merged**